internal/cli/pipeline_run.go

```diff
@@ -99,6 +99,7 @@ func (c *PipelineRunCommand) Run(args []string) int {
 			Application:  app.Ref(),
 			Workspace:    c.project.WorkspaceRef(),
 			TargetRunner: &pb.Ref_Runner{Target: &pb.Ref_Runner_Any{}},
+			Variables:    c.variables,
```

Comment:
I haven't dug deeper so it might be there, but I don't see the code where these variables are then handled?

Reply:
When you initialize a CLI client, we evaluate variables and load them onto the base client to be made available to any CLI clients:

https://github.com/hashicorp/waypoint/blob/b7c98b16f8959bc63d26ec88c313792408364fcc/internal/cli/base.go#L289-L299

Configuring a basic CLI client gets these variables attached to it (i.e. `c.variables`)

https://github.com/hashicorp/waypoint/blob/b7c98b16f8959bc63d26ec88c313792408364fcc/internal/cli/base_init.go#L95-L104

CLI commands like `waypoint up`, `waypoint deploy`, etc use DoApp to run app operations directly, and when that happens, the job client gets those variables attached to the job template:

https://github.com/hashicorp/waypoint/blob/b7c98b16f8959bc63d26ec88c313792408364fcc/internal/client/job.go#L22-L42

The `waypoint pipeline run` CLI is a bit different because it sends the job template off to the server to be used in the job generater when resolving pipeline steps to queued job requests. So the fix here is including those init CLI variables on the job template!

https://github.com/hashicorp/waypoint/blob/1698c1df4e167f794dcc044e9dd0dde23f2d5a03/internal/cli/pipeline_run.go#L97-L103